### ISSUE TITLE
if-then-else not cut transparent or some other problem?

### ISSUE THREAD

**Issue by Jean-Luc-Picard-2021** [trealla-prolog/trealla#130]
Now I was currious whether if-then-else is cut transparent,
and can reach a cleanup handler. I got a discrepancy:
```
/* Trealla Prolog 2.11.2 */
?- catch((call_cleanup(between(1,5,N), throw(error)), (between(1,5,M) -> !)), E, true).
   N = 1, M = N
;

/* SWI-Prolog 9.1.5 */
?- catch((call_cleanup(between(1,5,N), throw(error)), (between(1,5,M) -> !)), E, true).
E = error.
```

**Comment by infradig** [comment]
The cut does what it's supposed to do. Problem is the top-level sees the
call_cleanup registered handler and is too dumb, it thinks it's a normal
choice-point so gives the prompt. I'll look into it.

**Comment by Jean-Luc-Picard-2021** [comment]
I made an git pull and make of Trealla Prolog. 
When I press semicolon `;`  I see the throw:
```
/* Trealla Prolog 2.11.3 */
?- catch((call_cleanup(between(1,5,N), throw(error)), (between(1,5,M) -> !)), E, true).
   N = 1, M = N
;  E = error.
```
That is strange, since the handler should have been removed by the cut.
So after the cut there should not be any choice points around,
including not be any cleanup handlers around. The cleanup handlers 

should have already fired by the cut. Just like here, no more choice
points, except that the query result will not be success, but rather some throw.
```
?- between(1,5,N), (between(1,5,M)->!).
   N = 1, M = N.
```
